cli/coverage.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+#![allow(unused)]
+
+use crate::futures::SinkExt;
+use crate::futures::StreamExt;
+use crate::tokio_util;
+use deno_core::ErrBox;
+use serde::Deserialize;
+use serde::Serialize;
+use url::Url;
+
+pub struct CoverageCollector {
+  socket: tokio_tungstenite::WebSocketStream<tokio::net::TcpStream>,
+}
+
+// TODO(caspervonb) do not hard-code message ids.
+// TODO yield then await each command response after sending the request.
```

Comment:
I think we'll need to implement something like inspector agent that can automatically track proper ids

Reply:
Don't need anything particularly fancy for this in this case, just increment by one per message is fine.

Long term for the REPL etc yeah we'll definitively want a full blown inspector client.

Reply:
Right, but then you need to receive messages from inspector for each sent message and I believe inspector can send other messages as well which should be ignored.

Reply:
> Right, but then you need to receive messages from inspector for each sent message

The two TODO's tie together somewhat, e.g each socket send will loop awaiting the next response, breaking when a response with the sent id is received then error checking the result.

> I believe inspector can send other messages as well which should be ignored.

It can.

Enabling a domain (e.g calling the Runtime.enable method) will enable event notifications for that domain. Events do not carry identifiers however so this doesn't interfere with the above.